iothub/device/src/Transport/AmqpIot/AmqpIotTransport.cs

```diff
@@ -177,13 +182,20 @@ private async Task<ClientWebSocket> CreateClientWebSocketAsync(Uri websocketUri,
                     {
                         // Configure proxy server
                         websocket.Options.Proxy = webProxy;
-                        Logging.Info(this, $"{nameof(CreateClientWebSocketAsync)} Setting ClientWebSocket.Options.Proxy");
+                        if (Logging.IsEnabled)
+                            Logging.Info(this, $"{nameof(CreateClientWebSocketAsync)} Setting ClientWebSocket.Options.Proxy");
                     }
                 }
                 catch (PlatformNotSupportedException)
                 {
                     // .NET Core 2.0 doesn't support proxy. Ignore this setting.
-                    Logging.Error(this, $"{nameof(CreateClientWebSocketAsync)} PlatformNotSupportedException thrown as .NET Core 2.0 doesn't support proxy");
+                    if (Logging.IsEnabled)
+                        Logging.Error(this, $"{nameof(CreateClientWebSocketAsync)} PlatformNotSupportedException thrown as .NET Core 2.0 doesn't support proxy");
+                }
+
+                if (_amqpTransportSettings.WebSocketKeepAlive.HasValue)
```

Comment:
Interesting!

Reply:
how so?

Reply:
I had not tried the Nullable.HasValue before. Seems like a useful property to have so that we longer have to do null and default value checks.